src/api/python/hidra/utils/utils_general.py

```diff
@@ -371,6 +372,35 @@ def create_sub_dirs(dir_path, subdirs, dirs_not_to_create=()):
         raise OSError
 
 
+class TempFile:
+    def __init__(self, temp_file, target_path):
+        self.temp_file = temp_file
+        self.target_path = str(target_path)
```

Comment:
is the parameter `target_path` expected to be non-str?
is the data member `temp_file` expected to be non-str?

Wondering why there is `str()` on one line and not on the other

Reply:
I guess `temp_file` is a `file` object?
So `TempFile` objects are a (file object, file path) structure, got it.

Reply:
Continue to be confused by the absence of type information

Reply:
I'd suggest docstrings here (or for the class definition)

Reply:
What about keeping `target_path` as is, and only converting it to `string` when required? At the moment the `string` form is only used in the call to `os.rename`.